src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+		userId = player.UserId,
+		eventType = "Purchase",
+		currency = "Coins",
+		itemId = "100 Coins"
+	})
+	```
+]=]
+function AnalyticsService:LogResourceEvent(data: ResourceEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogResourceEvent - data is required")
+	assert(typeof(data.userId) == "number", "AnalyticsService.LogResourceEvent - userId is required")
+	assert(typeof(data.eventType) == "string", "AnalyticsService.LogResourceEvent - eventType is required")
+	assert(
+		table.find(RESOURCE_EVENT_TYPES, data.eventType),
+		"AnalyticsService.LogResourceEvent - eventType " .. data.eventType .. " is invalid. Please define it in RESOURCE_EVENT_TYPES"
+	)
+	assert(data.itemId, "AnalyticsService.LogResourceEvent - itemId is required")
+	assert(data.currency, "AnalyticsService.LogResourceEvent - currency is required")
+	assert(
+		table.find(CURRENCIES, data.currency),
+		"AnalyticsService.LogResourceEvent - currency type is invalid"
+	)
+	assert(
+		GameAnalytics.EGAResourceFlowType[data.flowType],
+		"AnalyticsService.LogResourceEvent - flow type is invalid"
+	)
+	assert(
+		typeof(data.amount) == "number",
+		"AnalyticsService.LogResourceEvent - amount is required"
+	)
+	
+	GameAnalytics:addResourceEvent(data.userId, {
+		-- FlowType is Sink by default
+		flowType = data.flowType,
+		currency = data.currency,
+		amount = data.amount,
+		itemType = data.eventType,
+		itemId = data.itemId
+	})
+end
+
+function AnalyticsService:LogError(data: ErrorEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogError - data is required")
+	assert(data.userId, "AnalyticsService.LogError - userId is required")
+	assert(data.message, "AnalyticsService.LogError - message is required")
+	assert(
+		data.severity == nil or GameAnalytics.EGAErrorSeverity[data.severity],
+		"AnalyticsService.LogError - error severity type is invalid"
+	)
+	
+	local errorSeverity: string = data.severity or GameAnalytics.EGAErrorSeverity.Error
+	
+	GameAnalytics:addErrorEvent(data.userId, {
+		message = data.message,
+		severity = GameAnalytics.EGAErrorSeverity[errorSeverity]
+	})
+end
+
+--[=[
+	Used to track player progression (example: player score in a mission or level)
+	A progression can have up to 3 levels (example: Mission 1, Location 1, Level 1)
+	
+	Example:
+	```lua
+	AnalyticsService:LogProgression({
+		userId = player.UserId,
+		status = "Start",
+		progression01 = "Mission X",
+		progression02 = "Location X",
+		score = 100 -- Started with score of 100
+	})
+	AnalyticsService:LogProgression({
+		userId = player.UserId,
+		status = "Complete",
+		progression01 = "Mission X",
+		progression02 = "Location X",
+		score = 400 -- Completed the mission with a score of 400
+	})
+	```
+	
+	For more information on progression events, refer to [GameAnalytics docs](https://docs.gameanalytics.com/integrations/sdk/roblox/event-tracking?_highlight=teleportdata#progression) on progression.
+]=]
+function AnalyticsService:LogProgression(data: ProgressionEvent)
```

Comment:
Add param tag to the moonwave block.
@param data ProgressionEvent -- ProgressionEvent type